api/lib/jwt_service.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# This module provides JWT token encoding and decoding functionality.
+module TokenAuthorization
+  # encoding data into a jwt token
+
+  def self.encode(payload, exp = 24.hours.from_now)
+    payload[:exp] = exp.to_i
+    begin
+      JWT.encode(payload, ENV.fetch('SECRET_KEY'))
+    rescue JWT::EncodeError => e
+      raise StandardError, "Error encoding payload: #{e.message}"
```

Comment:
Instead of raising an error, you can return nil, similar to the `decode` method.